### PR TITLE
add `queue_max_size` parameter for socket write opertaions

### LINKED ISSUE #3: Add ability to limit packet queue
Right now, `Socket` protocol have unlimited `_packets` queue, this tends to resource problems in a «bad» case. Maybe as a default `unlimited` is good enough, but this definitely should be configurable.

### PR DESCRIPTION
**Pull request by pohmelie**
closes #3

### CODE REVIEW

File: asyncudp/__init__.py [hunk 1]
```diff
@@ -67,7 +69,7 @@ async def create_socket(local_addr=None, remote_addr=None):
 
     loop = asyncio.get_running_loop()
     transport, protocol = await loop.create_datagram_endpoint(
-        _SocketProtocol,
+        partial(_SocketProtocol, queue_max_size=queue_max_size),
```
Comment (eerimoq) [inline asyncudp/__init__.py]: No need to use functools if I'm not mistaken.

```suggestion
        lambda: _SocketProtocol(queue_max_size),
```

File: asyncudp/__init__.py [hunk 2]
```diff
@@ -58,7 +59,7 @@ async def recvfrom(self):
         return await self._protocol.recvfrom()
 
 
-async def create_socket(local_addr=None, remote_addr=None):
+async def create_socket(local_addr=None, remote_addr=None, *, queue_max_size=0):
```
Comment (eerimoq) [inline asyncudp/__init__.py:62]: What value does the `*` add? I mean, I read that at least one core Python thinks it's confusing to new (and old) Python users =)

Comment (pohmelie) [reply asyncudp/__init__.py:62]: `*` do not add any values. In python it is just a mark, that rest of arguments are keyword-only arguments.

**Comment by eerimoq** [comment]
Merged.

**Pull request closed without merging**